File: client/src/app/_models/auction.ts
export interface Auction {
  title: string;
  description: string;
  currentPrice: number;
  currentBidder: string;

  duration: number;
  isFinished: boolean;
  auctionWinner: string;
  vehicleId: number;
  userId: number;
}
export class AuctionClass implements Auction {
  title: string = '';
  description: string = '';
  price: string = '';
  vehicleId: number = null;
  userId: number = null;
  currentPrice: number = 1;
  currentBidder: string = '';
  duration: number = null;
  isFinished: boolean = false;
  auctionWinner: string = '';
  constructor() {}
}
export class DurationClass {
  durationList: any = [1, 2, 5, 7];
}

File: API/Models/Auction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Models
{
    public class Auction
    {
        public long AuctionId { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }


        public float CurrentPrice { get; set; }

        public string CurrentBidder { get; set; }

        public DateTime Start { get; set; }
        public DateTime End { get; set; }

        public bool IsFinished { get; set; }

        public string AuctionWinner { get; set; }

        public ICollection<Offer> Offers { get; set; }


        public long UserId { get; set; }
        public User User { get; set; }


        public long VehicleId { get; set; }
        public Vehicle Vehicle { get; set; }
    }
}